lib/src/facts/aix/processor_resolver.cc

```diff
@@ -0,0 +1,117 @@
+#include <internal/facts/aix/processor_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/util/scope_exit.hpp>
+
+#include <cstdint>
+#include <stdexcept>
+#include <odmi.h>
+#include <sys/cfgodm.h>
+
+using namespace std;
+
+static const char* odm_error_string() {
+    static char* msg;
+    int result = odm_err_msg(odmerrno, &msg);
+    if (result < 0) {
+        return "failed to retrieve ODM error message";
+    } else {
+        return msg;
+    }
+}
+
+#define check_odm_return(var, msg) if (reinterpret_cast<intptr_t>(var) == -1) { throw runtime_error((boost::format(msg ": %1%") % odm_error_string()).str()); }
+
+struct physical_processor
+{
+     string type;
+     long long frequency;
+     int smt_threads;
+     bool smt_enabled;
+};
+
+namespace facter { namespace facts { namespace aix {
+
+    processor_resolver::data processor_resolver::collect_data(collection& facts)
+    {
+        auto result = posix::processor_resolver::collect_data(facts);
+
+        if (odm_initialize() < 0) {
+            throw runtime_error((boost::format("failed to initialize ODM: %1%") % odm_error_string()).str());
+        }
+        leatherman::util::scope_exit(odm_terminate);
+
+        CLASS_SYMBOL PdDv_class = odm_mount_class((char *)"PdDv");
```

Comment:
C++-style casts are preferred over C-style casts.  `const_cast<char*>("PdDv")` is a clearer indication that this terrible, terrible function isn't const-correct (at least one hopes it doesn't invoke UB by writing to these strings).